gensim/models/word2vec.py

```diff
@@ -1956,8 +1956,7 @@ def _load_specials(self, *args, **kwargs):
         if not hasattr(self.wv, 'vectors_lockf') and hasattr(self.wv, 'vectors'):
             self.wv.vectors_lockf = np.ones(1, dtype=REAL)
         if not hasattr(self, 'random'):
-            # use new instance of numpy's recommended generator/algorithm
-            self.random = np.random.default_rng(seed=self.seed)
+            self.random = np.random.RandomState(self.seed)
```

Comment:
Regresses https://github.com/RaRe-Technologies/gensim/issues/2782.

Reply:
fixed ab53e62

Reply:
> Wouldn't that preclude us from using any new API functions from numpy? (new = introduced since oldest-supported-numpy)

It doesn't have to. Where we can easily replicate the new API functions, we should do so (as @menshikh-iv has done with gensim.utils.default_rng function). Where we can't (so far, there are no such cases, but there may be in the future), we could do something like:

```python
def np_cool_new_function(*args, **kwargs):
    try:
        fn = np.cool_new_function
    except AttributeError:
        raise RuntimeError("your numpy version is too old, please upgrade")
    else:
        return fn(*args, **kwargs)
```

Reply:
Perhaps. Although this smells of re-implementing dependency resolution internally, which would be extra hell.

But I guess as long as `setup.py` requires an actual working version of numpy (not just `oldest-supported-numpy`), it should be OK.

But requiring some dependency version in `setup.py`, and then raising RuntimeErrors suggesting that version is too old, would not be OK.

Reply:
> But I guess as long as setup.py requires an actual working version of numpy (not just oldest-supported-numpy), it should be OK.

The problem with going that way is that it prevents us from building against oldest-supported-numpy. We then have to reinvent _that_ particular wheel in our build system infrastructure.

Reply:
Let's see if I got this right:

|                    |                      | numpy API (setup.py) |                   |
|--------------------|----------------------|----------------------|-------------------|
|                    |                      | oldest-supported-numpy | 1.17.0            |
| numpy ABI (wheels) | oldest-support-numpy | this #3231                | current `develop` |
|                    | 1.17.0               | ?                    | desired, https://github.com/RaRe-Technologies/gensim/issues/3226#issuecomment-917364050           |

Building against the oldest numpy *we actually support* makes more sense to me than building against oldest-supported-numpy and then throwing RuntimeError because it's too old.

@mpenkov what exactly was the problem with that?

Reply:
It's not exactly code, it's [build system configuration](https://github.com/RaRe-Technologies/gensim/blob/96add7f315735e9579608111a0d1fc72df8e4e0b/.github/workflows/build-wheels.yml#L24-L108).

We're essentially replicating [this](https://github.com/scipy/oldest-supported-numpy/blob/master/setup.cfg) with slightly different business logic.

Reply:
OK. Reading the comment in your `.yml` link, that sounds exactly like what we want = the "desired" state. Why does that not work?